### PR TITLE
feat: generate call descriptors and use startApiCall

### PR DESCRIPTION
**Pull request by noahdietz**
This refactors the generated client implementation to use the `startApiCall` invoker from `gax`. This also includes generation of extra call descriptors in the config files that described the call type and all properties.

This is an entirely internal refactor and does not affect the public surface.

### CODE REVIEW

**Comment by noahdietz** [comment]
Verified with some sample tests that things still work with these changes :D

File: .github/workflows/bazel_example.yml [hunk 1]
```diff
@@ -2,9 +2,9 @@ name: Bazel Build
 
 on:
   push:
-    branches: [ main ]
+    branches: [ main, v2-dev ]
```
Comment (bshaffer) [inline .github/workflows/bazel_example.yml:5]: We'd want to remove this before merging into main, yes?

Comment (bshaffer) [reply .github/workflows/bazel_example.yml:5]: Now I'm thinking we want to keep it, since we're keeping the `v2-dev` branch even after this is merged.

Comment (noahdietz) [reply .github/workflows/bazel_example.yml:5]: yeah I think we should keep it if we are keeping the branch.

**Comment by noahdietz** [comment]
Looks like I need to update the bazel goldens after catching up with `main`

**Comment by noahdietz** [comment]
Closing this PR as we've decided to move these changes to the separate v2 client. @dwsupplee will be migrating the work.

**Pull request closed without merging**